synapse/tests/test_cortex.py

```diff
@@ -101,6 +101,70 @@ async def test_cortex_divert(self):
             self.len(2, [n for n in nodes if n.ndef[0] == 'inet:fqdn'])
             self.len(2, await core.nodes('ou:org +#faz'))
 
+            # functions that don't return a generator
+            storm = '''
+            function x() {
+                $lst = $lib.list()
+                [ ou:org=* ou:org=* +#cat ]
+                $lst.append($node)
+                fini { return($lst) }
+            }
+
+            divert $lib.true $x()
+            '''
+            await self.asyncraises(s_exc.BadArg, core.nodes(storm))
+            self.len(2, await core.nodes('ou:org +#cat'))
+
+            storm = '''
+            function x() {
+                $lst = $lib.list()
+                [ ou:org=* ou:org=* +#dog ]
+                $lst.append($node)
+                fini { return($lst) }
+            }
+
+            [ inet:fqdn=vertex.link inet:fqdn=woot.com ]
+
+            divert $lib.true $x()
+            '''
+            await self.asyncraises(s_exc.BadArg, core.nodes(storm))
+            self.len(4, await core.nodes('ou:org +#dog'))
```

Comment:
Note that the function still gets invoked multiple times unless we detected/raised in setArgv

Reply:
I guess we could make a `storm:node:genr` type and make an explicit check against the AstType we get inside setArgv()?  Does that seem too hacky?